packages/runtime/src/utils/storeScrollState.ts

```diff
@@ -0,0 +1,64 @@
+type ScrollPosition = {
+  x: number
+  y: number
+}
```

Comment:
Consider this to get better auto completion + highlight that every key doesn't necessarily have a value:

```suggestion
interface ScrollPosition {
  x: number
  y: number
}

type ScrollPositionKey = '__window' | string

type ScrollPositions = Partial<Record<ScrollPositionKey, ScrollPosition>>
```